typespec/Applications.Core/applications.tsp

```diff
@@ -37,15 +37,29 @@ using OpenAPI;
 
 namespace Applications.Core;
 
-@doc("Radius Application resource")
-model ApplicationResource is TrackedResource<ApplicationProperties> {
+@doc("Concrete tracked resource types can be created by aliasing this type using a specific property type.")
+@Azure.ResourceManager.Private.armResourceInternal(TProperties)
+@includeInapplicableMetadataInPayload(false)
+model TrackedResourceRequired<
+  TProperties extends {}
+> extends Azure.ResourceManager.Foundations.TrackedResourceBase {
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "template"
+  @doc("The resource-specific properties for this resource.")
+  @visibility("read", "create")
+  @extension("x-ms-client-flatten", true)
+  properties: TProperties;
+
   @doc("The application name")
   @path
   @key("applicationName")
   @segment("applications")
+  @visibility("read", "create")
   name: ResourceNameString;
 }
 
+@doc("Radius Application resource")
+model ApplicationResource is TrackedResourceRequired<ApplicationProperties>;
+
 @doc("Application properties")
 model ApplicationProperties {
   @doc("The status of the asynchronous operation.")
```

Comment:
I have created https://github.com/radius-project/radius/issues/6944 to track this for other resources

Reply:
Still checking with typespec team on how to move the TrackResourceRequired stuff to a common place to share with all resources. We can go ahead with this for this PR and the common stuff will be implemented while fixing #6944